lemminx-liberty/src/main/java/io/openliberty/tools/langserver/lemminx/services/XmlReader.java

```diff
@@ -0,0 +1,89 @@
+package io.openliberty.tools.langserver.lemminx.services;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.util.logging.Logger;
+
+import javax.xml.stream.XMLEventReader;
+import javax.xml.stream.XMLInputFactory;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.events.XMLEvent;
+
+public class XmlReader {
+    private static final Logger LOGGER = Logger.getLogger(XmlReader.class.getName());
+    private static XmlReader instance;
+
+    public static XmlReader getInstance() {
```

Comment:
Why is this a singleton? Couldn't we just have this be a utility class with static methods? Just wondering what the reasons were for it being a singleton.